### ISSUE TITLE
Pose estimation Error

### ISSUE THREAD

**Issue by srinimd2005** [wg-perception/linemod#37]
Hi
I am having a problem of pose deviation error. I there any possible way to correct the error. For example the pose for Coke can is above the coke can.

**Comment by fivef** [comment]
Perhaps I have the same problem:

![ork_colorhist_cut](https://cloud.githubusercontent.com/assets/1077593/24855515/1ff9da64-1de2-11e7-9815-1ac8049d3ba0.png)

My objects were trained via ork_capture.
The origin of the stl is at the bottom not in the center. @nlyubova Should we change https://github.com/wg-perception/reconstruction/blob/master/apps/mesh_object to create stls with the origin centered?

**Comment by JimmyDaSilva** [comment]
A quick shift of the mesh in blender should be enough

**Comment by fivef** [comment]
Yes shifting manually works but what do you think of modifying https://github.com/wg-perception/reconstruction/blob/master/apps/mesh_object to move the origin e.g. with a meshlab filter?

**Comment by nlyubova** [comment]
The origin of the mesh should be in the center. But this shift between the recognized object and the real one (as you have) can be also because of other parameters. For example, the frames used in the config file should be the one for the RGB camera and not for the depth one. Is it the case for you?

**Comment by fivef** [comment]
So you say ```depth_frame_id: 'camera_depth_optical_frame'``` in my detection.ros.ork should be set to the RGB camera frame? 

After shifting the origin to the center the alignment improves but still is not perfect. I will further try to improve this by tuning `icp_dist_min` and `px_match_min`.

Another question: Is it ok to use the topic `camera/rgb/image_rect_color` and `camera/depth_registered/image_raw` as my Xtion driver doesn't give me the image_raw topic. Does reconstruction and detection work with the raw data + the camera_info or is it ok to input the rectified  `image_rect_color`?

**Comment by nlyubova** [comment]
Try to replace the depth topic to the RGB topic and you will see if there is any difference.
 
For detection, you can use rectified image.

**Comment by fivef** [comment]
Setting the `depth_frame_id` to `camera_rgb_optical_frame` and playing with the topics didn't help.

**Comment by MengDes** [comment]
Hi i think i got the same problem ，have you solve the question ？

**Comment by TONG369** [comment]
> Hi i think i got the same problem ，have you solve the question ？
Hello, has your problem been solved? I met the same problem as you. How can I solve it?

**Comment by MengDes** [comment]
> Hi i think i got the same problem ，have you solve the question ？
> Hello, has your problem been solved? I met the same problem as you. How can I solve it?

the mesh's origin that ork provide  not in center, you can ajdust the center,or find some new mesh file , then train your own template

**Comment by TONG369** [comment]
Thanks  for your reply.
Do you know which file was modified?




At 2019-10-15 10:35:59, "MengDes" <notifications@github.com> wrote:


Hi i think i got the same problem ，have you solve the question ？
Hello, has your problem been solved? I met the same problem as you. How can I solve it?

the mesh's origin that ork provide not in center, you can ajdust the center,or find some new mesh file , then train your own template

**Comment by MengDes** [comment]
> Thanks for your reply. Do you know which file was modified? At 2019-10-15 10:35:59, "MengDes" <notifications@github.com> wrote: Hi i think i got the same problem ，have you solve the question ？ Hello, has your problem been solved? I met the same problem as you. How can I solve it? the mesh's origin that ork provide not in center, you can ajdust the center,or find some new mesh file , then train your own template — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

it 's been a long time ,i am quiet not sure,you can search on internet

**Comment by TONG369** [comment]
thank you very much  !!!






At 2019-10-15 12:08:27, "MengDes" <notifications@github.com> wrote:


Thanks for your reply. Do you know which file was modified? At 2019-10-15 10:35:59, "MengDes" notifications@github.com wrote: Hi i think i got the same problem ，have you solve the question ？ Hello, has your problem been solved? I met the same problem as you. How can I solve it? the mesh's origin that ork provide not in center, you can ajdust the center,or find some new mesh file , then train your own template — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

it 's been a long time ,i am quiet not sure,you can search on internet
